core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -465,35 +480,31 @@ public void processJournalEntry(JournalEntry entry) throws IOException {
 
   @Override
   public void start(Boolean isPrimary) throws IOException {
-    if (isPrimary) {
-      // Only initialize root when isPrimary because when initializing root, BlockMaster needs to
-      // write journal entry, if it is not primary, BlockMaster won't have a writable journal.
-      // If it is secondary, it should be able to load the inode tree from primary's checkpoint.
-      mInodeTree.initializeRoot(SecurityUtils.getOwnerFromLoginModule(),
-          SecurityUtils.getGroupFromLoginModule(), Mode.createFullAccess().applyDirectoryUMask());
-      String rootUfsUri = Configuration.get(PropertyKey.MASTER_MOUNT_TABLE_ROOT_UFS);
-      Map<String, String> rootUfsConf =
-          Configuration.getNestedProperties(PropertyKey.MASTER_MOUNT_TABLE_ROOT_OPTION);
-      long rootUfsMountId = IdUtils.ROOT_MOUNT_ID;
-      try {
-        // NOTE, we currently treat root inode specially and add it to inode tree manually.
-        mMountTable.add(new AlluxioURI(MountTable.ROOT), new AlluxioURI(rootUfsUri),
-            rootUfsMountId, MountOptions.defaults().setShared(
-                UnderFileSystemUtils.isObjectStorage(rootUfsUri) && Configuration
-                    .getBoolean(PropertyKey.UNDERFS_OBJECT_STORE_MOUNT_SHARED_PUBLICLY))
-                .setProperties(rootUfsConf));
-      } catch (FileAlreadyExistsException e) {
-        // This can happen when a primary becomes a standby and then becomes a primary.
-        LOG.info("Root has already been mounted.");
-      } catch (InvalidPathException e) {
-        throw new IOException("Failed to mount the default UFS " + rootUfsUri, e);
-      }
-    }
-    // Call super.start after mInodeTree is initialized because mInodeTree is needed to write
-    // a journal entry during super.start. Call super.start before calling
-    // getExecutorService() because the super.start initializes the executor service.
     super.start(isPrimary);
     if (isPrimary) {
+      LOG.info("Starting fs master as primary");
+
+      InodeDirectory root = mInodeTree.getRoot();
+      if (root == null) {
+        try (JournalContext context = createJournalContext()) {
```

Comment:
I wanted to include the root inode as a part of the zero-state for a file system master. Unfortunately, this isn't possible because we need to remember security information about previous starts of the file system master. The root inode is owned by the user that first starts the Alluxio master. If the master is stopped and another user tries to start it, we need to remember the original user so that we can complain that the new user doesn't have permission.